torch/csrc/profiler/combined_traceback.h

```diff
@@ -58,10 +58,17 @@ struct TORCH_API CapturedTraceback : public c10::GatheredContext {
   int traversePython(visitproc visit, void* arg);
   int clearPython();
 
+  void record_user_defined_frame(
```

Comment:
nit: can we match the method style for this class, `recordUserDefinedFrame`

Reply:
Let's have this API take a Frame object directly rather than name/stage so that it doesn't become specific to record function.